graph/vertex/impl/VertexImpl.java

```diff
@@ -51,7 +46,7 @@ public String toString() {
     @Override
     public boolean equals(Object object) {
         if (this == object) return true;
-        if (object == null || getClass() != object.getClass()) return false;
+        if (!(object instanceof VertexImpl)) return false; // instanceof includes null check
```

Comment:
now that a `ThingVertex` can either a `Read` or `Write`, we need the equality check to treat them as equal! This bug sent me around the houses for a while!